platform/cog-platform-fdo.c

```diff
@@ -357,7 +358,7 @@ registry_global (void               *data,
                                               name,
                                               &zxdg_shell_v6_interface,
                                               version);
-        assert (wl_data.xdg_shell != NULL);
+        g_assert_nonnull (wl_data.xdg_shell);
```

Comment:
The `g_assert_*` family of functions is only intended for use in unit tests. It prints the result in a way that would be useful when logging test failures. You should just use normal `g_assert` instead. That's the only form of assertion that should be used outside of your tests.

Reply:
Why should only `g_assert` used in unit tests? The [documentation](https://developer.gnome.org/glib/stable/glib-Testing.html) does not mention anything about that. If something, it *encourages* using the extended `g_assert_*` functions to get better diagnostics:

> In addition to the traditional g_assert(), the test framework provides an extended set of assertions for comparisons: g_assert_cmpfloat(), g_assert_cmpint(), g_assert_cmpuint(), g_assert_cmphex(), g_assert_cmpstr(), and g_assert_cmpmem(). The advantage of these variants over plain g_assert() is that the assertion messages can be more elaborate, and include the values of the compared entities.

Reply:
That's the documentation of the test framework. In unit tests, you would indeed want to prefer one of the more specific asserts, and avoid using `g_assert`. Outside tests, it's the opposite. Anyway, it's just a rule that GNOME projects use... I thought it was written down somewhere, maybe in the programming guidelines or someplace, but I'm not sure.